lib/jsdom/living/helpers/cors-settings.js

```diff
@@ -0,0 +1,47 @@
+"use strict";
+
+// https://html.spec.whatwg.org/multipage/urls-and-fetching.html#cors-settings-attributes
+
+const states = {
+  NoCORS: { keywords: [null] },
+  // canonical keyword is `keywords[0]` (anonymous)
+  Anonymous: { keywords: ["anonymous", ""], mode: "cors", credentialsMode: "same-origin" },
+  UseCredentials: { keywords: ["use-credentials"], mode: "cors", credentialsMode: "include" }
+};
+
+const invalidValueDefault = states.Anonymous;
+const missingValueDefault = states.NoCORS;
+const keywordsMap = new Map();
+
+Object.entries(states).forEach(stateKV => {
+  const state = states[stateKV[0]];
+  const { keywords } = stateKV[1];
+  keywords.forEach(keyword => {
+    if (keyword !== null) {
+      keywordsMap.set(keyword.toLowerCase(), state);
+    }
+  });
+});
```

Comment:
This all looks a bit over-engineered. We generally represent constants with just strings. What do you think of this:

```suggestion
const invalidValueDefault = "anonymous";
const missingValueDefault = "no-cors";

const keywordsMap = {
  __proto__: null,
  anonymous: "anonymous",
  "": "anonymous",
  "use-credentials": "use-credentials"
};
```

The `keywordsMap` could be a `Map` if you wish to use that instead. `getKeyword` will no longer be necessary after this since the state is represented already by the the canonical keyword. For the "No CORS" state representing it with `null` seems to suffice.

The `mode` and `credentialsMode` information should not exist in this file. Instead, they should be encoded as part of a function implementing https://html.spec.whatwg.org/multipage/urls-and-fetching.html#create-a-potential-cors-request, which probably goes into `resource-loader.js`.